src/aws-manager.js

```diff
@@ -139,10 +125,172 @@ class AwsManager {
     this.reportAmiUsage = series.amiUsage.reporter(influx);
   }
 
+  async init(){
+    try {
+      let rawJson = await this.spotRequestContainer.read('internal-provisioner-data');
+      let data = JSON.parse(rawJson);
+      // This is mainly for debugging
+      fs.writeFileSync('internal-data.json', data);
+      this.__spotRequestIdCache = data.managedSpotRequests || []; 
+      this.__internalState = data.internallyTrackedSpotRequests || [];
+    } catch (err) {
+      if (err.code !== 'BlobNotFound') {
+        throw err;
+      }
+      this.__spotRequestIdCache = [];
+      this.__internalState = [];
+    }
+  }
+
+  async saveAwsManagerInternalState() {
+    // We use this for both of the current internal caches... let's just use
+    // the same code for both
+    function expire(item) {
+      let diff = Date.now() - item.created;
+      // If we've known about this for more than 5 days, evict it, otherwise
+      // keep it.  5 days is a good amount since we force kill things even if
+      // they're running after 4 days.  This lets us have a full extra day
+      if (diff > 1000 * 60 * 60 * 24 * 5) {
+        return false;
+      }
+      return true;
+    }
+
+    this.__spotRequestIdCache = this.__spotRequestIdCache.filter(expire);
```

Comment:
So they are, I'm 99% sure.  The last time we talked about this, it was a security issue around worker-ids, if I'm recalling correctly.  In that case, the fear was that an unprivileged worker type could get a worker id for a wrong instance type.  For security issues like that, I think it was the right call to not do something because of that edge case.

This case is a lot different, however.  The worst case is that if we had a spot instance reuse, we'd count an instance for the wrong worker type for provisioning and if the wrong worker type were deleted, we might accidentally kill it.  This isn't a security issue, but sadly, it's basically the only way for us to link a spot request to a worker type in a reliable way :(

We'd also likely only be dealing with one or two instance if this happened, and we evict SRIDs after 5 days, so we'd need to have a signficant amount of reuse within the same region within the same 5 day period, and the worst case is that we are off on provisioning by at most the amount of capacity of the instances with reused spot instance request ids.

I can live with this risk, how about you?